TripKitUI/controller/TKSectionedAlertViewModel.swift

```diff
@@ -0,0 +1,151 @@
+      .map { self.alertItems(from: $0) }
+      .map { self.alertSections(from: $0) }
+      .share(replay: 1)
+  }()
+  
+  public init(region: SVKRegion) {
+    self.region = region
+  }
+  
+  // MARK: -
+  
+  private func alertItems(from decoded: [API.AlertMapping]) -> [String: [AlertGroup]] {
+    var alertGroupsByModes: [String: [AlertGroup]] = [:]
+    var tmp: [AlertGroup] = []
+    decoded.forEach { mapping in
+      let mode = mapping.modeIdentifier ?? "Unknown mode"
+      if let existing = alertGroupsByModes[mode] {
+        var currentRouteIds = existing.map { $0.routeId }
+        tmp = existing
+        mapping.routes?.forEach {
+          if let index = currentRouteIds.index(of: $0.identifier) {
+            var currentGroup = tmp[index]
+            currentGroup.alerts.append(mapping)
+            tmp[index] = currentGroup
+            alertGroupsByModes[mode] = tmp
+          } else {
+            let newGroup = AlertGroup(routeId: $0.identifier, alerts: [mapping])
+            tmp.append(newGroup)
+            currentRouteIds.append($0.identifier)
+            alertGroupsByModes[mode] = tmp
+          }
+        }
+      } else {
+        var groups: [AlertGroup] = []
+        mapping.routes?.forEach {
+          let newGroup = AlertGroup(routeId: $0.identifier, alerts: [mapping])
+          groups.append(newGroup)
+        }
+        alertGroupsByModes[mode] = groups
+      }
+    }
+    
+    return alertGroupsByModes
+    
+//    var itemsByModes: [String: [String]] = [:]
+//    decoded.forEach { (mapping) in
+//      let mode = mapping.modeIdentifier ?? "Unknown mode"
+//      if let existing = itemsByModes[mode] {
+//        let existingSet = Set(existing)
+//        let newSet = Set(mapping.routeIds)
+//        itemsByModes[mode] = Array(existingSet.union(newSet))
+//      } else {
+//        itemsByModes[mode] = mapping.routeIds
+//      }
+//    }
+//    return itemsByModes
+  }
+  
+//  private func alertSections(from itemsByMode: [String: [String]]) -> [AlertSection] {
+//    var sections: [AlertSection] = []
+//
+//    itemsByMode.forEach { (key, value) in
+//      let items = value.map { AlertItem(routeId: $0) }
+//      let section = AlertSection(header: key, items: items)
+//      sections.append(section)
+//    }
+//
+//    return sections
+//  }
+  
+  private func alertSections(from alertGroupsByMode: [String: [AlertGroup]]) -> [AlertSection] {
+    var sections: [AlertSection] = []
+    
+    alertGroupsByMode.forEach { (key, value) in
+      let items = value.map { AlertItem(alertGroup: $0) }
+      let section = AlertSection(header: key, items: items)
+      sections.append(section)
+    }
+    
+    return sections
+  }
+  
+}
+
+// MARK: -
+
+extension API.AlertMapping {
+  
+  var routeIds: [String] {
+    guard let routes = self.routes else { return [] }
+    return routes.map { $0.identifier }
+  }
+  
+}
+
+extension API.Route {
+  
+  var identifier: String {
```

Comment:
This is my desperate attempt to make sure I have a valid id/label/tag to group alerts since all properties in `API.Route` are all optional...